src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -663,14 +671,8 @@ private void finish(@NonNull Result r, @CheckForNull Throwable t) {
             onEndBuilding();
         } finally {  // Ensure this is ALWAYS removed from FlowExecutionList
             FlowExecutionList.get().unregister(new Owner(this));
-            listener = null;
-        }
-
-        try {
-            StashManager.maybeClearAll(this, /* or move up before closing getListener()? */new LogTaskListener(LOGGER, Level.FINE));
-        } catch (IOException | InterruptedException x) {
-            LOGGER.log(Level.WARNING, "failed to clean up stashes from " + this, x);
         }
+        completeAsynchronousExecution();
```

Comment:
It seems unusual to have this one line outside of the `try`/`finally` that surrounds the rest of the method. Should it go into the `finally` block in case something in `try` throws? Or is the `try` block not actually necessary in practice and so it does not really matter?

Reply:
Probably makes sense to put it there as well.